src/aspire/orientation/__init__.py

```diff
@@ -1,38 +1,503 @@
 import logging
+import math
+import numpy as np
+import scipy.sparse as sparse
+
+from aspire import config
+
+from aspire.basis.polar_2d import PolarBasis2D
+from aspire.source import ArrayImageSource
+from aspire.utils.coor_trans import common_line_from_rots
+from aspire.utils.matlab_compat import m_reshape
 
 logger = logging.getLogger(__name__)
 
 
-class Orient3D:
+class CLOrient3D:
     """
-    Define a base class for estimating 3D orientations
+    Define a base class for estimating 3D orientations using common lines methods
     """
-    def __init__(self, src):
+    def __init__(self, src, n_rad=None, n_theta=None, n_check=None):
         """
-        constructor of an object for estimating 3D orientations
+        Initialize an object for estimating 3D orientations using common lines
+
+        :param src: The source object of 2D denoised or class-averaged imag
+        :param n_rad: The number of points in the radial direction
+        :param n_theta: The number of points in the theta direction
+        :param n_check: For each image/projection find its common-lines with
+            n_check images. If n_check is less than the total number of images,
+            a random subset of n_check images is used.
```

Comment:
This is a good point.  But it becomes more complicated globally if we want to compare the results with Matlab version. In the `matlab_compat.py` we reimplemented `randi` and `randn`  so that the same random numbers can be used for other modules such regenerating orientation angles and 2d images. Here we need to set seed for numpy. They are different things. I included the option of seed explicitly for numpy in commit 8775255 so that we will have option to select it and unit tests can be recovered easily too.  For the global problem we should address it in other PRs.

Reply:
k, please roll back the bug added in the https://github.com/ComputationalCryoEM/ASPIRE-Python/commit/8775255be3869dae7d8b2f158cd0d4765516e776 and I'll close this. We don't need to make a code change, there is an issue for it.

Reply:
Reverted the changes in the commit 5bcb18b.

Reply:
confirmed, thanks